runtime/sema/check_path_expression.go

```diff
@@ -19,40 +19,59 @@
 package sema
 
 import (
+	"regexp"
+
 	"github.com/onflow/cadence/runtime/ast"
 	"github.com/onflow/cadence/runtime/common"
 )
 
 func (checker *Checker) VisitPathExpression(expression *ast.PathExpression) ast.Repr {
 
-	ty, err := CheckPathLiteral(expression)
-	checker.report(err)
+	ty := CheckPathLiteral(
+		expression.Domain.Identifier,
+		expression.Identifier.Identifier,
+		func(errFunc func(err *ast.PathExpression) error) {
+			checker.report(errFunc(expression))
+		},
+	)
 
 	return ty
 }
 
-func CheckPathLiteral(expression *ast.PathExpression) (Type, error) {
+func CheckPathLiteral(domainString, identifier string, report func(func(err *ast.PathExpression) error)) Type {
```

Comment:
Maybe instead of requiring a path expression to produce the error, generalize it to just what is needed to produce the error, the range. That way a caller can e.g. pass an empty location range